src/modals/createChannel.ts

```diff
@@ -0,0 +1,37 @@
+import {
+    Modal,
+    MessageActionRow,
+    ModalActionRowComponent,
+    TextInputComponent,
+    ModalSubmitInteraction,
+} from 'discord.js';
+
+export const data = new Modal()
+    .setCustomId('createChannelModal')
+    .setTitle('Create New Channel')
+    .addComponents(
+        new MessageActionRow<ModalActionRowComponent>().addComponents(
+            new TextInputComponent()
+                .setCustomId('channelName')
+                .setLabel('Channel name')
+                .setStyle('SHORT')
+        ),
+        new MessageActionRow<ModalActionRowComponent>().addComponents(
+            new TextInputComponent()
+                .setCustomId('channelDescription')
+                .setLabel('Channel description')
+                .setStyle('SHORT')
+        )
+    );
+export async function execute(interaction: ModalSubmitInteraction) {
+    const channelName = interaction.fields.getTextInputValue('channelName');
+    const channelDescription =
+        interaction.fields.getTextInputValue('channelDescription');
+
+    // Create new channel, or assign it if user is trying to create existing channel
+    console.log({ channelName, channelDescription });
```

Comment:
Is this line needed?

Reply:
I have stopped working on create new channel modal after we agreed that channel creation will have further rules, it's technically unused code for now, hence just the log, but it's ready to be finished when channel creation criteria are finalized